spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -203,10 +203,14 @@ protected Object handleRequestMessage(Message<?> message) {
 			throw new MessageHandlingException(
 					message, "Failed to write data to BigQuery tables in message handler: " + this, e);
 		}
-		catch (InterruptedException | ExecutionException | TimeoutException e) {
+		catch (ExecutionException | TimeoutException e) {
 			throw new MessageHandlingException(
 					message, "Failed to wait for BigQuery Job to complete in message handler: " + this, e);
 		}
+		catch (InterruptedException e) {
+			Thread.currentThread().interrupt();
```

Comment:
Is this a potentially silent failure? How do we let the caller know that the write failed?

Reply:
I think the calling code is supposed to handle it - the article listed in the description said if you can't throw the checked exception at least do this.

Reply:
The we shouldn't catch exception, right?

Reply:
It's a checked exception - you have to handle it.

Reply:
or declare it thrown

Reply:
It's an overridden method - so you can't change the signature.

Reply:
Would it still make sense to throw the `MessageHandlingException` after the `interrupt()` in order not to introduce a breaking change?